pallets/account/src/lib.rs

```diff
@@ -0,0 +1,308 @@
+pub mod pallet {
+    use super::*;
+
+    #[pallet::pallet]
+    pub struct Pallet<T>(PhantomData<T>);
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// The overarching event type
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+        /// The Currency for managing Evm account assets
+        type Currency: Mutate<Self::AccountId>;
+        /// Default evm address to account id conversion
+        type DefaultAddressMapping: AddressMapping<Self::AccountId>;
+        /// Default account id to evm address conversion
+        type DefaultAccountMapping: AccountMapping<Self::AccountId>;
+        /// The Signature verification implementation to use for checking claims
+        /// Note: the signature type defined by this will be used as parameter in pallet's extrinsic
+        type ClaimSignature: ClaimSignature<AccountId = Self::AccountId, Address = EvmAddress>;
+
+        // /// Weight information for the extrinsics in this module
+        // type WeightInfo: WeightInfo;
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        /// AccountId has mapped
+        AccountIdHasMapped,
+        /// Eth address has mapped
+        EthAddressHasMapped,
+        /// Bad signature
+        BadSignature,
+        /// Invalid signature
+        InvalidSignature,
+    }
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// EVM Account claimed.
+        /// Double Mapping b/w native and evm address created
+        AccountClaimed {
+            account_id: T::AccountId,
+            evm_address: EvmAddress,
+        },
+    }
+
+    /// Native accounts for EVM address
+    /// NativeAccounts: EvmAddress => Option<AccountId>
+    #[pallet::storage]
+    pub type NativeAccounts<T: Config> =
+        StorageMap<_, Twox64Concat, EvmAddress, T::AccountId, OptionQuery>;
+
+    /// EVM Addresses for native accounts
+    /// EvmAccounts: AccountId => Option<EvmAddress>
+    #[pallet::storage]
+    pub type EvmAccounts<T: Config> =
+        StorageMap<_, Twox64Concat, T::AccountId, EvmAddress, OptionQuery>;
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Claim account mapping between Substrate accounts and EVM accounts.
+        /// Ensure no prior mapping exists for evm address.
+        ///
+        /// - `evm_address`: The address to bind to the caller's account
+        /// - `signature`: A signature generated by the address to prove ownership
+        #[pallet::call_index(0)]
+        #[pallet::weight(0)]
+        pub fn claim_evm_account(
+            origin: OriginFor<T>,
+            evm_address: EvmAddress,
+            signature: SignatureOf<T>,
+        ) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+            // make sure no prior mapping exists
+            Self::enure_no_mapping(&who, &Some(evm_address))?;
```

Comment:
This is a precautionary move since the action of changing H160's underlying SS58 address is very "sensitive" in nature. 

Once balances are mapped they can be used interchangeably, for example CEX withdrawal in evm account by using native address in CEX that is connected to users evm account. So if user changed his mappings third party (dapps, saved address, another user, even smart contracts) can still be using his/her old mapped address that can be problematic.

The wallet accounts are dispensable in nature so even if user has a reason to change the mapping, he/she can always create a new account and transfer all the funds to it until we allow it change.
We can enable this later once we have feedbacks from user that this is a MUST have feature otherwise we play safe.

Reply:
IMO update mapping brings more edge cases to handle and troubles to indexers than benefits. A fixed mapping should be enough to achieve the goal of lowering the bar for UX.